options/options.h

```diff
@@ -0,0 +1,73 @@
+/*********************                                                        */
+/*! \file
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Florian Lonsing, Makai Mann
+ ** This file is part of the pono project.
+ ** Copyright (c) 2019, 2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file LICENSE in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief
+ **
+ **
+ **/
+
+#pragma once
+
+#include "utils/ponoresult.h"
+#include "utils/exceptions.h"
+#include "options/defaults.h"
+
+/*************************************** Options class
+ * ************************************************/
+// Meant to be used as a singleton class
+
+namespace pono {
+
+class PonoOptions
+{
+
+  // TODO: for now, all class members (i.e., Pono options) are public
```

Comment:
We could also just make this a struct. For this kind of object, I'm not opposed to keeping everything public. What are your thoughts @lonsing and @ahmed-irfan?

Reply:
struct sounds good.

Reply:
on a second thought, maybe class would be more clean and safe.